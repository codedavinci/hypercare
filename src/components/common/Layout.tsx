import { ReactNode } from "react";

type LayoutProps = {
  children: ReactNode;
};

export default function Layout({ children }: LayoutProps) {
  return (
    <div
      className="m-10 grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4"
      role="group"
    >
      {children}
    </div>
  );
}

Layout.displayName = "Layout";
