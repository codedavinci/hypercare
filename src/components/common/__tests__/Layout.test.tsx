import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Layout from "../Layout";

describe("Layout Component", () => {
  const sampleChildren = (
    <>
      <div>Child 1</div>
      <div>Child 2</div>
    </>
  );

  it("should render correctly and match snapshot", () => {
    const { asFragment } = render(<Layout>{sampleChildren}</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render children correctly", () => {
    render(<Layout>{sampleChildren}</Layout>);
    expect(screen.getByText("Child 1")).toBeInTheDocument();
    expect(screen.getByText("Child 2")).toBeInTheDocument();
  });

  it("should apply the correct classes", () => {
    render(<Layout>{sampleChildren}</Layout>);
    const layoutElement = screen.getByRole("group"); // We need to give the container a role for testing
    expect(layoutElement).toHaveClass("m-10");
    expect(layoutElement).toHaveClass("grid");
    expect(layoutElement).toHaveClass("grid-cols-1");
    expect(layoutElement).toHaveClass("sm:grid-cols-2");
    expect(layoutElement).toHaveClass("md:grid-cols-3");
    expect(layoutElement).toHaveClass("lg:grid-cols-4");
    expect(layoutElement).toHaveClass("gap-4");
  });
});
