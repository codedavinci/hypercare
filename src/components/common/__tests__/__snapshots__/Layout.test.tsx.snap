// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-10 grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4"
    role="group"
  >
    <div>
      Child 1
    </div>
    <div>
      Child 2
    </div>
  </div>
</DocumentFragment>
`;
